### ISSUE TITLE
CuTeDSL kernel name prefix leaks into later compilations

### ISSUE THREAD

**Issue by drisspg** [NVIDIA/cutlass#3452]
### Bug

A `_name_prefix` supplied to one `@cute.kernel` launch remains on CuTeDSL's singleton DSL object and is reused by later kernel launches that do not specify a prefix. As a result, profiler traces depend on compilation order and can label unrelated kernels as the earlier operation. And all my beautiful names are ruined 😭

### Minimal reproducer

```python
import json
from pathlib import Path

import torch
from cutlass import Float32, cute


class FirstOperation:
    @cute.jit
    def __call__(self, x: cute.Tensor, stream):
        self.kernel(x, _name_prefix="first_prefix").launch(
            grid=(1, 1, 1), block=(32, 1, 1), stream=stream
        )

    @cute.kernel
    def kernel(self, x: cute.Tensor):
        thread, _, _ = cute.arch.thread_idx()
        if thread < cute.size(x):
            x[thread] = x[thread] + 1.0


class SecondOperation:
    @cute.jit
    def __call__(self, x: cute.Tensor, stream):
        self.kernel(x).launch(
            grid=(1, 1, 1), block=(32, 1, 1), stream=stream
        )

    @cute.kernel
    def kernel(self, x: cute.Tensor):
        thread, _, _ = cute.arch.thread_idx()
        if thread < cute.size(x):
            x[thread] = x[thread] * 2.0


def compile_kernel(entrypoint, artifact_name):
    fake_x = cute.runtime.make_fake_compact_tensor(
        Float32, (32,), stride_order=(0,), assumed_align=16
    )
    fake_stream = cute.runtime.make_fake_stream(use_tvm_ffi_env_stream=True)
    return cute.compile[cute.EnableTVMFFI](
        entrypoint, fake_x, fake_stream, _name_prefix=artifact_name
    )


first = compile_kernel(FirstOperation(), "first_artifact")
second = compile_kernel(SecondOperation(), "second_artifact")
x = torch.ones(32, device="cuda")

trace_path = Path("/tmp/cute_prefix_repro.json")
with torch.profiler.profile(activities=[torch.profiler.ProfilerActivity.CUDA]) as profile:
    first(x)
    second(x)
torch.cuda.synchronize()
profile.export_chrome_trace(str(trace_path))

trace = json.loads(trace_path.read_text())
for event in trace["traceEvents"]:
    if event.get("cat") == "kernel":
        print(event["name"])
```

Run with the internal CuTeDSL cache disabled to rule out a stale artifact:

```console
$ CUTE_DSL_NO_CACHE=1 python cute_prefix_repro.py
first_prefix_kernel_cutlass_kernel___main__FirstOperation_object_at__tensorptrf32gmemalign16o321_0
first_prefix_kernel_cutlass_kernel___main__SecondOperation_object_at__tensorptrf32gmemalign16o321_0
```

The second line is the issue: `SecondOperation` never sets `first_prefix`, but receives it anyway. Giving the second launch an explicit `_name_prefix="second_prefix"` makes its name correct.

### Expected behavior

A launch's `_name_prefix` should be scoped to that launch/compilation. A later kernel with no explicit prefix should use its normal generated name, not a prefix left by an earlier kernel.

### Environment

- `nvidia-cutlass-dsl==4.6.0.dev0`
- Python 3.13.12
- PyTorch `2.14.0.dev20260811+cu132`
- CUDA runtime 13.2
- NVIDIA GB300, compute capability 10.3

**Comment by anakinxc** [comment]
Taking a look now

**Comment by drisspg** [comment]
So the pubic one .set_name_prefix has the same problem - but it seems it might have been fixed in 4.6.2 

so feel free to close

**Comment by anakinxc** [comment]
Yup, verified `set_name_prefix` in latest release fixed this

@Junkai-Wu please close this. Thanks
